yazi-config/src/theme/filetype.rs

```diff
@@ -1,21 +1,36 @@
-use std::path::Path;
+use std::str::FromStr;
 
+use anyhow::bail;
 use serde::{Deserialize, Deserializer};
-use yazi_shared::MIME_DIR;
+use yazi_shared::fs::File;
 
 use super::{Color, Style, StyleShadow};
 use crate::Pattern;
 
 pub struct Filetype {
+	pub is:    FiletypeIs,
 	pub name:  Option<Pattern>,
 	pub mime:  Option<Pattern>,
 	pub style: Style,
 }
 
 impl Filetype {
-	pub fn matches(&self, path: &Path, mime: Option<&str>) -> bool {
-		let is_dir = mime == Some(MIME_DIR);
-		self.name.as_ref().is_some_and(|n| n.match_path(path, is_dir))
+	pub fn matches(&self, file: &File, mime: Option<&str>) -> bool {
+		let b = match self.is {
+			FiletypeIs::None => true,
+			FiletypeIs::Block => file.cha.is_block_device(),
+			FiletypeIs::Exec => file.cha.is_exec(),
+			FiletypeIs::Fifo => file.cha.is_fifo(),
+			FiletypeIs::Link => file.cha.is_link(),
+			FiletypeIs::Orphan => file.cha.is_orphan(),
+			FiletypeIs::Sock => file.cha.is_socket(),
+			FiletypeIs::Sticky => file.cha.is_sticky(),
+		};
+		if !b {
+			return false;
+		}
+
+		self.name.as_ref().is_some_and(|n| n.match_path(&file.url, file.is_dir()))
```

Comment:
Doesn't seem to actually work... `b` is ignored if true

Reply:
Hmm have you set `name` or `mime` for it?

Reply:
Oh ok, now I get it. I was used to my old implementation which didn't require that

Reply:
Yeah I want to use it as an extension to the existing `name` and `mime` - further restricting files that match either `name` or `mime`, rather than adding a new rule. This is more conducive to later expanding other attributes. 😉